aspnet/fundamentals/logging.rst

```diff
@@ -3,117 +3,171 @@
 
-You may see more than one log statement per web request you make in your browser, since most browsers will make multiple requests (i.e. for the favicon file) when attempting to load a page. Note that the console logger displayed the log level (``info`` in the image above) followed by the category (``[Catchall Endpoint]``), and then the message that was logged.
+.. literalinclude:: logging/sample/src/TodoApi/Controllers/TodoController.cs
+  :language: c#
+  :start-after: snippet_LoggerDI
+  :end-before: endregion
+  :emphasize-lines: 4,7,10
 
-The call to the log method can utilize a format string with named placeholders (like `{path}`). These placeholders are populated in the order in which they appear by the args values passed into the method call. Some logging providers will store these names along with their mapped values in a dictionary that can later be queried. In the example below, the request path is passed in as a named placeholder:
+You can then create logs by calling methods named ``Log`` + the severity level of the log to write, as in this example.
 
-.. literalinclude:: logging/sample/src/TodoApi/Startup.cs
+.. literalinclude:: logging/sample/src/TodoApi/Controllers/TodoController.cs
   :language: c#
-  :lines: 54
-  :dedent: 16
+  :start-after: snippet_CallLogMethods
+  :end-before: endregion
+  :emphasize-lines: 3,7
 
-In your real world applications, you will want to add logging based on application-level, not framework-level, events. For instance, if you have created a Web API application for managing To-Do Items (see :doc:`/tutorials/first-web-api`), you might add logging around the various operations that can be performed on these items.
+At this point, your logging method calls will be successful, but you won't see log output anywhere until you configure at least one logging provider.
 
-The logic for the API is contained within the `TodoController`, which uses :doc:`dependency-injection` to request the services it requires via its constructor. Ideally, classes should follow this example and use their constructor to `define their dependencies explicitly <http://deviq.com/explicit-dependencies-principle/>`_ as parameters. Rather than requesting an `ILoggerFactory` and creating an instance of `ILogger` explicitly, `TodoController` demonstrates another way to work with loggers in your application - you can request an `ILogger<T>` (where `T` is the class requesting the logger).
+Configure logging providers in your ``Startup`` class. Include an ``ILoggerFactory`` parameter in the method signature of the ``Configure`` method, and DI provides an object that you call methods on to add and configure providers. To add a provider, you can call an ``AddProvider`` method and pass in a provider instance, but there's typically an extension method that simplifies the syntax. The following example adds the providers for sending log output to the console and to the Visual Studio Debug window.
```

Comment:
I present them in this order on purpose.  One of the causes of confusion in the original article is a lack of clarity about the distinction between the logging API you call to create logs and the logging providers that do something with data from the logging API method calls.  My intention in showing the logger API first is to help make clear the distinction between the logger API and the logging providers.  You can call log methods and everything appears to work, but the data only shows up somewhere when you set up a logging provider.